### ISSUE TITLE
RequestConnectSpring bug

### ISSUE THREAD

**Issue by GoogleFrog** [ZeroK-RTS/Zero-K-Infrastructure#1261]
I added the ability to spectate games just by clicking on a button in a list. This uses RequestConnectSpring without being in the battleroom. I did this for a battle, the battle ended and I started a singleplayer game (most likely not relevant). During the SP game, Spring started and I found out that I was a player in the battle started from the same battleroom that I had sent RequestConnectSpring to in order to spectate the game before. This game was exited and the same thing occurred shortly later when the game was started again.

In short, the battleroom seemed to think I was a player even though I had never joined the battle and the server never indicated that I had joined the battle.

Here is an unuseful infolog. [infolog.txt](https://github.com/ZeroK-RTS/Zero-K/files/485606/infolog.txt)
